alpha/apps/kaltura/lib/storage/kFileSyncUtils.class.php

```diff
@@ -253,6 +292,56 @@ public static function file_put_contents ( FileSyncKey $key , $content , $strict
 		self::setPermissions($fullPath);
 		self::createSyncFileForKey($rootPath, $filePath,  $key , $strict , !is_null($res), false, md5($content), kPathManager::getStorageProfileIdForKey($key));
 		self::encryptByFileSyncKey($key, kPathManager::getStorageProfileIdForKey($key));
+		self::setInSmallFileCache($key, $content);
+	}
+
+	protected static function setInSmallFileCache($key, $content)
+	{
+		$contentToPut = gzcompress($content) ??  $content;
+		$compressedContentToPut = gzcompress($contentToPut);
+		$contentToPut = $compressedContentToPut ? $compressedContentToPut : $contentToPut;
+
+		//If the file is under a specific threshold in size, save it to redis cache
+		$contentSize = strlen($contentToPut);
+		$fileSyncFilesToCache = kConf::get(self::FILE_SYNC_TYPES_TO_CACHE, kConfMapNames::RUNTIME_CONFIG, array());
+		$maxFileSizeToCache = kConf::get(self::MAX_FILE_SIZE_TO_CACHE, kConfMapNames::RUNTIME_CONFIG);
+		if (!$fileSyncFilesToCache || !$maxFileSizeToCache)
+		{
+			return;
+		}
+		if ($contentSize <= $maxFileSizeToCache && in_array("{$key->object_type}_{$key->object_sub_type}", $fileSyncFilesToCache))
```

Comment:
why not do this check on type before doing gzip?
gzip is CPU and type verification is not - if type is not to be stored in cache, we can avoid the gzip time/cpu.